rust/pattern/conjunctable.rs

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+use std::fmt;
+use std::fmt::Display;
+
+use crate::enum_getter;
+use crate::pattern::*;
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum Conjunctable {
+    Negation(()),
+    Variable(Variable),
+}
+
+impl Conjunctable {
+    enum_getter!(into_variable, Variable, Variable);
```

Comment:
Quite frankly, this macro may have become obsolete for TypeQL. There are only two places left that use it, so it may be scrapped altogether in the end.

More to the point, ideally it would just take the enum variant name and its type and autogenerate the `into_X()` conversions based on the variant name. There might be a proc macro crate that does that through some black magic, but to the best of my knowledge token concatenation (which would be needed to create an `into_X` token from just `X`) is currently unstable.